test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/ConfigurationManagerTelemetryTests.cs

```diff
@@ -200,6 +206,7 @@ public static TheoryData<ConfigurationManagerTelemetryTheoryData<OpenIdConnectCo
                         { TelemetryConstants.MetadataAddressTag, OpenIdConfigData.AADCommonUrl },
                         { TelemetryConstants.IdentityModelVersionTag, IdentityModelTelemetryUtil.ClientVer },
                         { TelemetryConstants.OperationStatusTag, TelemetryConstants.Protocols.Automatic },
+                        { TelemetryConstants.ConfigurationSourceTag, blocking ? TelemetryConstants.Protocols.ConfigurationSourceRetriever : TelemetryConstants.Protocols.ConfigurationSourceUnknown }
```

Comment:
This confuses me, can you help me understand?

Reply:
This is due to differences on blocking and non-blocking paths, in places where telemetry is reported we sometimes don't know which source will be used. Alternative to this approach is to refactor more of the main flow.

Reply:
Let's make that difference extremely clear in doc strings on the option for blocking at least

Reply:
Refactored this a bit so that only Manual and LKG operation have Unknown source.

Reply:
Reverted. I made it more complicated. Added more comments for clarity.